thresher/serializers.py

```diff
@@ -199,14 +199,87 @@ def to_internal_value(self, data):
                                
 
 # A serializer for a highlight group
-class HighlightGroupSerializer(serializers.ModelSerializer):
-    # a custom field containing all the questions and answers
-    questions = serializers.ListField(child=GenericSubmittedAnswerField())
+# class HighlightGroupSerializer(serializers.ModelSerializer):
+#     # a custom field containing all the questions and answers
+#     questions = serializers.ListField(child=GenericSubmittedAnswerField())
+#     offsets = JSONSerializerField()
+
+#     class Meta:
+#         model = HighlightGroup
+#         fields = ('tua', 'offsets', 'questions')
+
+#     def create(self, validated_data):
+#         # Get the answers nested models
+#         answers = validated_data.pop('questions')
+
+#         # Remove the force_insert if it's there
+#         validated_data.pop('force_insert', None)
+
+#         # create the highlight group first
+#         highlight_group = HighlightGroup.objects.create(**validated_data)
+
+#         # Add the highlight group model to the kwargs and save
+#         for answer in answers:
+#             model = answer['class']
+#             kwargs = answer['data']
+#             # Add the highlight group instance to the kwargs
+#             kwargs['highlight_group'] = highlight_group
+            
+#             # There is a special case if it's a checklist,
+#             # Because of the many to many relationship, this needs to be saved differently
+#             if model == CLSubmittedAnswer:
+#                 # Get the answers:
+#                 answers = kwargs.pop('answer')
+#                 # first create the CLSubmitted answer
+#                 submission = CLSubmittedAnswer.objects.create(**kwargs)
+#                 # Now add the answers
+#                 submission.answer.add(*answers)
+
+#             # For all other models, simply create the objects
+#             else:
+#                 model.objects.create(**kwargs)
+
+#         return highlight_group 
+
+class OffsetField(serializers.Field):
```

Comment:
Definitely the correct approach! However, I think right now this is looking to serialize the offsets in the following way:
{ "target":
              {"selector":
                              {"@type":"TextPositionSelector",
                               "offsets":[<start-index>, <end-index>],
                              }
                 }
}
This is not exactly what the W3C standard requires for the TextPositionSelector. Specifically, the TextPositionSelector is expect to have 2 fields, "start" with the start index and "end" with the end index. Refer to https://www.w3.org/TR/annotation-model/#text-position-selector for details. 

Something to consider here is that it's possible for the TUA to not be representable with a simple TextPositionSelector if it has multiple start and end indices. This is why we had talked about adding a new selector type, perhaps called "MultipleSpanPositionSelector", that allows for that. In which case I think we should go with something like:
{"offsets":
             [
              {"start":<start-index_1>, "end":<end-index_1>},
              {"start":<start-index_2>, "end":<end-index_2>},
              ....
              ]}
Maybe @nickbadams and @stefanv can give some input on that?